### PR TITLE
executor, parser: indicate analyze is auto or not in proto buffer

### LINKED ISSUE #34607: In gRPC message, indicate the Analyze is Auto Analyze or not
## Enhancement

As a pre-request of auto analyze flow control,  TiDB server needs to tell backend TiKV if the Analyze request is from Auto Analyze or user's manual ones.

### PR DESCRIPTION
**Pull request by BornChanger**
Signed-off-by: BornChanger <dawn_catcher@126.com>

### What problem does this PR solve?

Issue Number: close #34607

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
   Verified the new flag for auto analyze is passed in proto buffer as expected to TiKV side
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: executor/builder.go [hunk 1]
```diff
@@ -2582,13 +2590,20 @@ func (b *executorBuilder) buildAnalyzeColumnsPushdown(task plannercore.AnalyzeCo
 	failpoint.Inject("injectAnalyzeSnapshot", func(val failpoint.Value) {
 		startTS = uint64(val.(int))
 	})
+
+	isAutoFlag := uint64(0)
+
+	if autoAnalyze != "" {
```
Comment (zhangjinpeng1987) [inline executor/builder.go:2596]: Any test to cover this case?

Comment (BornChanger) [reply executor/builder.go:2596]: No test case for it yet. I tested it manually at TiKV server side.

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
.I think We can add ```with-real-tikv``` test cases to `tests`.

**Comment by BornChanger** [comment]
> .I think We can add `with-real-tikv` test cases to `tests`.

The flag is not hardly dependent on tikv side. There is no-op at tikv side if it does not recognize the flag.

**Pull request merged**